### ISSUE TITLE
Consider pulling NamedTuples magic into Columns

### ISSUE THREAD

**Issue by tshort** [JuliaComputing/IndexedTables.jl#33]
If `Columns` could pull in the magic from `NamedTuples`, it would make the API nicer. I could refer to `tbl.data.colX` and `tbl.index.a` instead of  `tbl.data.columns.colX` and `tbl.index.columns.a`.

**Comment by JeffBezanson** [comment]
Agreed. I think this will require dot overloading for a reasonable implementation though.

**Comment by shashi** [comment]
How about having `data(t, :colname)` and `index(t, :colname)` for now? Omitting the second argument will return the whole `Columns`.

Something like
```julia
@inline function data(t, name)
    getfield(t.data.columns, name)
end
```
Can be made inferable? (Not sure why I get a `Union` in `@code_warntype` despite using `getfield`)
